main.tf

```diff
@@ -290,3 +292,44 @@ resource "google_storage_bucket_iam_member" "api_s_account_storage_admin_on_proj
 
   depends_on = ["google_project_service.project_services"]
 }
+
+/******************************************
+  compute.networkUser role granted to API and GKE service accounts for GKE on shared VPC 
+ *****************************************/
+resource "google_project_iam_binding" "gke_shared_vpc_project" {
+  count = "${(var.shared_vpc != "" && contains(var.activate_apis, "container.googleapis.com") && length(compact(var.shared_vpc_subnets)) == 0) ? 1 : 0}"
+
+  role    = "roles/compute.networkUser"
+  project = "${var.shared_vpc}"
+  members = ["${compact(list(local.api_s_account_fmt, local.gke_s_account_fmt))}"]
+
+  depends_on = ["google_project_service.project_services"]
+}
+
+/******************************************
+  compute.networkUser role granted to API and GKE service accounts for GKE on shared VPC subnets
+ *****************************************/
+resource "google_compute_subnetwork_iam_binding" "gke_shared_vpc_subnets" {
+  count = "${(contains(var.activate_apis, "container.googleapis.com") && length(compact(var.shared_vpc_subnets)) != 0) ? length(var.shared_vpc_subnets) : 0}"
+
+  subnetwork = "${element(split("/", var.shared_vpc_subnets[count.index]), 5)}"
+  role       = "roles/compute.networkUser"
+  region     = "${element(split("/", var.shared_vpc_subnets[count.index]), 3)}"
+  project    = "${var.shared_vpc}"
+  members    = ["${compact(list(local.api_s_account_fmt, local.gke_s_account_fmt))}"]
+
+  depends_on = ["google_project_service.project_services"]
+}
+
+/******************************************
+  container.hostServiceAgentUser role granted to GKE service account for GKE on shared VPC
+ *****************************************/
+resource "google_project_iam_binding" "gke_host_agent" {
+  count = "${contains(var.activate_apis, "container.googleapis.com") ? 1 : 0}"
```

Comment:
This should also test that Shared VPC is actually being used. (Reuse the same boolean I mentioned in above comment.)